onnx/serialization.py

```diff
@@ -165,8 +167,43 @@ def deserialize_proto(self, serialized: bytes | str, proto: _Proto) -> _Proto:
         return google.protobuf.json_format.Parse(serialized, proto)
 
 
+class _TextualSerializer(ProtoSerializer):
+    """Serialize and deserialize the ONNX textual representation."""
+
+    supported_format = "onnxtxt"
+    file_extensions = frozenset({".onnxtxt"})
+
+    def serialize_proto(self, proto: _Proto) -> bytes:
+        text = onnx.printer.to_text(proto)  # type: ignore[arg-type]
+        return text.encode(_ENCODING)
+
+    def deserialize_proto(self, serialized: bytes | str, proto: _Proto) -> _Proto:
+        warnings.warn(
+            "The onnxtxt format is experimental. Please report any errors to the ONNX GitHub repository.",
+            stacklevel=2,
+        )
+        if not isinstance(serialized, (bytes, str)):
+            raise TypeError(
+                f"Parameter 'serialized' must be bytes or str, but got type: {type(serialized)}"
+            )
+        if isinstance(serialized, bytes):
+            text = serialized.decode(_ENCODING)
+        else:
+            text = serialized
+        if isinstance(proto, onnx.ModelProto):
+            return onnx.parser.parse_model(text)  # type: ignore[return-value]
+        if isinstance(proto, onnx.GraphProto):
+            return onnx.parser.parse_graph(text)  # type: ignore[return-value]
+        if isinstance(proto, onnx.FunctionProto):
+            return onnx.parser.parse_function(text)  # type: ignore[return-value]
+        if isinstance(proto, onnx.NodeProto):
+            return onnx.parser.parse_node(text)  # type: ignore[return-value]
```

Comment:
Just curious: why it needs `# type: ignore[return-value]` here?

Reply:
Looks like mypy doesn't understand NodeProto is a `_Proto`. I wonder if there's a better type we can use